### PR TITLE
Refactor and document `Git`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

**Comment by dawidd6** [comment]
@reitermarkus May I have a request for you? Please check official taps for the usages of the code you are changing next time.

**Comment by reitermarkus** [comment]
@dawidd, I have all taps installed and did in fact check. Seems like I still managed to miss this one usage, sorry.

File: Library/Homebrew/test/utils/git_spec.rb [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 require "utils/git"
 
-describe Git do
+describe Utils::Git do
   before do
+    described_class.clear_available_cache
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/utils/git_spec.rb:7]: @reitermarkus I think this change or similar may be breaking parallel tests. Seeing lots of exceptions on macOS and Linux (that go away on rerun) when they can't find Git: https://github.com/Homebrew/brew/runs/1025893539?check_suite_focus=true#step:13:129

Comment (reitermarkus) [reply Library/Homebrew/test/utils/git_spec.rb:7]: Fixed in https://github.com/Homebrew/brew/pull/8486.

**Comment by MikeMcQuaid** [comment]
@reitermarkus Another flaky test regression from this in `tap_spec.rb` it seems, I'm afraid: https://github.com/Homebrew/brew/actions/runs/228376100

**Pull request merged**